searchlib/src/main/java/com/yahoo/searchlib/rankingexpression/rule/FunctionReferenceContext.java

```diff
@@ -43,26 +43,32 @@ public FunctionReferenceContext(Map<String, ExpressionFunction> functions) {
 
     /** Create a context for a single serialization task */
     public FunctionReferenceContext(Map<String, ExpressionFunction> functions, Map<String, String> bindings) {
-        this(ImmutableMap.copyOf(functions), bindings);
+        this.functions = Map.copyOf(functions);
+        if (bindings != null)
+            this.bindings.putAll(bindings);
     }
 
+    /** @deprecated Use {@link #FunctionReferenceContext(Map, Map)} instead */
+    @Deprecated(forRemoval = true, since = "7")
     protected FunctionReferenceContext(ImmutableMap<String, ExpressionFunction> functions, Map<String, String> bindings) {
-        this.functions = functions;
-        if (bindings != null)
-            this.bindings.putAll(bindings);
+        this((Map<String, ExpressionFunction>)functions, bindings);
     }
 
-    private static ImmutableMap<String, ExpressionFunction> toMap(Collection<ExpressionFunction> list) {
-        ImmutableMap.Builder<String,ExpressionFunction> mapBuilder = new ImmutableMap.Builder<>();
+    private static Map<String, ExpressionFunction> toMap(Collection<ExpressionFunction> list) {
+        Map<String, ExpressionFunction> mapBuilder = new HashMap<>();
```

Comment:
The name 'mapBuilder' is confusing unless you know that it used to be an ImmutableMap.Builder. I suggest to use e.g. functionByName, name2Function, or simply functions.